gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/model/InternalMedia.java

```diff
@@ -158,13 +159,33 @@ public String getApplicationThumbnailUrl(MediaResolution mediaResolution) {
 
   @Override
   public SilverpeasFile getFile(final MediaResolution mediaResolution) {
-    String origOrThumbFileName = getFileName();
-    if (getType().isPhoto() && StringUtil.isDefined(mediaResolution.getThumbnailSuffix())) {
-      origOrThumbFileName = getId() + mediaResolution.getThumbnailSuffix();
+    if (StringUtil.isNotDefined(getFileName())) {
+      return SilverpeasFile.NO_FILE;
+    }
+    List<String> fileNames = new ArrayList<String>(2);
+    fileNames.add(getFileName());
+    if (getType().isPhoto()) {
+      String thumbnailSuffix = mediaResolution.getThumbnailSuffix();
+      if (StringUtil.isDefined(thumbnailSuffix)) {
+        final String originalFileExt = "." + FilenameUtils.getExtension(getFileName());
+        fileNames.set(0, getId() + thumbnailSuffix + originalFileExt);
+        fileNames.add(getId() + thumbnailSuffix + ".jpg");
+      }
+    }
+    SilverpeasFile file = SilverpeasFile.NO_FILE;
+    for (String fileName : fileNames) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "fileName" which hides the field declared at line 58. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS135)